09-redux/classworks/Volodkova_Anna/src/App.js

```diff
@@ -0,0 +1,77 @@
+import React, {useEffect, useState, useRef} from 'react';
+import {useRouteMatch} from "react-router-dom";
+import {useSelector, useDispatch} from 'react-redux';
+import Header from 'components/header'
+import Pages from 'pages';
+import {withRedditApi} from './hooks/useRedditApi';
+import {setPostsNull, setSearchValue, fetchPosts} from "./resourses/actions";
+
+function App(props) {
+  const dispatch = useDispatch();
+  const getSearchValue = useSelector((state) => state.value);
+  let getSubreddits = useSelector((state)=>state.posts);
+  console.log('getSubreddits from app.js', getSubreddits);
+
+  const [reactSubreddit, setReactSubreddit] = useState(null);
+  const [reactAbout, setReactAbout] = useState(null);
+
+
+  const getdata = async (name = 'r/react') => {
+    const {fetchReddit} = props;
+    const data = await fetchReddit(`/${name}/hot`).then(res => res.json());
+    setReactSubreddit(data);
+    const about = await fetchReddit(`/${name}/about`).then(res => res.json());
+    setReactAbout(about);
+  };
+
+
+  useEffect(() => {
+    if (props.isToken) {
+      getdata();
+      dispatch(setPostsNull());
+    }
+  }, [props.isToken]);
+
+  const prevSubUrl = useRef();
+  let match = useRouteMatch("/subreddit/:subredditUrl");
+  console.log('Match', match);
+
+  useEffect(() => {
+    if (match && match.params.subredditUrl !== prevSubUrl.current) {
+      prevSubUrl.current = match.params.subredditUrl;
+      //getSubreddits(match.params.subredditUrl);
+      getdata(`r/${match.params.subredditUrl}`);
+      //setReactSubreddits(null);
+      dispatch(setPostsNull());
+    }
+  }, [match]);
+
+
+  const handleOnChangeInput = (event) => {
```

Comment:
we use Redux now to avoid passing props from the top component inside down. but you still pass it. you should have handlers in the components they are used by